ota-plus-web/app/com/advancedtelematic/controllers/Data.scala

```diff
@@ -0,0 +1,29 @@
+package com.advancedtelematic.controllers
+
+import java.time.Instant
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+
+import scala.collection.Seq
+
+object Data {
+
+  final case class FeedResource(createdAt: Instant, _type: String, resource: JsValue)
+
+  implicit val feedResourceWrites: Writes[FeedResource] = Json.writes[FeedResource]
+
+  private def literalStringReads(s: String): Reads[String] = (_: JsValue) => JsSuccess(s)
+
+  def feedResourceReads(_type: String): Reads[FeedResource] = (
+    (__ \ "createdAt").read[Instant] and
+    literalStringReads(_type) and
+    __.read[JsValue]
+  )(FeedResource.apply _)
```

Comment:
I would suggest to remove `literalStringReads` and add `FeedResource.ofType(t: String)(createdAt: Instant, resource: JsValue)

Reply:
Nice 👍 Updated.